### ISSUE TITLE
[Android] [SentryHttpClientException] Exclusions

### ISSUE THREAD

**Issue by barry-irvine** [getsentry/sentry-java#3244]
### Integration

sentry-android-okhttp

### Build System

Gradle

### AGP Version

8.2.2

### Proguard

Enabled

### Version

7.4.0

### Steps to Reproduce

I'm getting a number of 504 errors from `https://res.cloudinary.com` that I want to suppress. I am trying to suppress the error results from this domain when I add the Sentry interceptor. E.g.

```
  OkHttpClient.Builder().addInterceptor(SentryOkHttpInterceptor(failedRequestTargets = 
    listOf("^(?!(.*.cloudinary.com))"))).build()
```

Am I defining the regex incorrectly? I can't find any decent documentation on this.

### Expected Result

No errors are returned for given domain name.

### Actual Result

Sentry console filled with `SentryHttpClientException: HTTP Client Error with status code: 504` for this domain.

**Comment by markushi** [comment]
Hi @barry-irvine, thanks for reaching out!

`failedRequestTargets` can either be a plain string or an regex. But there's no need to define regex groups aka `()`. So in your case this should work fine:
```kotlin
OkHttpClient.Builder().addInterceptor(
    SentryOkHttpInterceptor(failedRequestTargets = listOf("^.*\\.cloudinary\\.com"))
).build()
```

**Comment by courville** [comment]
Same in my application, it is flooded with `mechanism SentryOkHttpInterceptor`, `SentryHttpClientException
HTTP Client Error with status code: 504`, see https://github.com/nova-video-player/aos-AVP/issues/1156

**Comment by markushi** [comment]
@courville thanks for chiming in here, do you have `failedRequestTargets` configured in your app?

**Comment by barry-irvine** [comment]
@markushi I tried it as you said and now have the following:

    @Provides
    @CommonOkHttp
    fun provideCommonOkHttpClient(): OkHttpClient =
        OkHttpClient.Builder()
            .addInterceptor(
                SentryOkHttpInterceptor(failedRequestTargets = listOf("^.*\\.cloudinary\\.com"))
            )
            .build()
            
  But I'm still seeing 504s on for example:

GET https://res.cloudinary.com/dtljonz0f/image/upload/c_fill,g_auto,h_240,w_240/gc-v1/new-york/Big-bus-tour-2.jpg

**Comment by barry-irvine** [comment]
Oh wait. I see your modified your answer after I implemented 🤦 . I'll give it another go

**Comment by romtsn** [comment]
@barry-irvine did you manage to verify whether the updated answer works for your case?

**Comment by courville** [comment]
> @courville thanks for chiming in here, do you have `failedRequestTargets` configured in your app?
> 
> This behavior is likely due to changes introduced in 7.1.0(see [changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md#710)), where the SDK would start reporting errors, which it previously didn't.

thanks for the feedback and https://github.com/nova-video-player/aos-Video/commit/930b65bb924fd4a52be3ff2229993090d7ccebb4 did the trick.

This was quite a change of behavior and unexpected.

Sorry for the noise.

**Comment by kahest** [comment]
@courville Apologies for the confusion - 7.1.0 fixed an issue where before that versoin, errors wouldn't be reported even though we communicated that they would. In general we don't change behavior like this outside of major version bumps.

**Comment by markushi** [comment]
Closing this for now, if you think there's still an issue simply re-open this ticket.

**Comment by barry-irvine** [comment]
@markushi I've finally got the updated code in production and can _still_ see the 504 cloudinary errors in Sentry.

This was the final implementation. Did I do something wrong?

    @Provides
    @CommonOkHttp
    fun provideCommonOkHttpClient(): OkHttpClient =
        OkHttpClient.Builder()
            .addInterceptor(
                SentryOkHttpInterceptor(failedRequestTargets = listOf("^((http|https)://)?(?!.*\\.?cloudinary\\.com).*$"))
            ).build()
            
And yet I see a 504 error for "https://res.cloudinary.com/dtljonz0f/image/upload/c_fill,g_auto,h_333,w_250/gc-v1/las-vegas/Favorite-Bistro7206_0.jpg" in the console. It looks like we get the 504s when the user's device is offline and Coil (the library we use for image processing) doesn't even bother to send the image request to Cloudinary because it knows they're offline so it just returns a 504 straightaway.

**Comment by angusholder** [comment]
> [@markushi](https://github.com/markushi) I've finally got the updated code in production and can _still_ see the 504 cloudinary errors in Sentry.
> 
> This was the final implementation. Did I do something wrong?
> 
> ```
> @Provides
> @CommonOkHttp
> fun provideCommonOkHttpClient(): OkHttpClient =
>     OkHttpClient.Builder()
>         .addInterceptor(
>             SentryOkHttpInterceptor(failedRequestTargets = listOf("^((http|https)://)?(?!.*\\.?cloudinary\\.com).*$"))
>         ).build()
> ```
> 
> And yet I see a 504 error for "https://res.cloudinary.com/dtljonz0f/image/upload/c_fill,g_auto,h_333,w_250/gc-v1/las-vegas/Favorite-Bistro7206_0.jpg" in the console. It looks like we get the 504s when the user's device is offline and Coil (the library we use for image processing) doesn't even bother to send the image request to Cloudinary because it knows they're offline so it just returns a 504 straightaway.

For anyone who was having this issue of non-existent 504s being reported to Sentry, it was just fixed in 8.40.0 here: #5299
